tests/tracer/test_tracer.py

```diff
@@ -1365,13 +1365,15 @@ def test_manual_keep(tracer, test_spans):
         s.set_tag(MANUAL_KEEP_KEY)
     spans = test_spans.pop()
     assert spans[0].metrics[SAMPLING_PRIORITY_KEY] is USER_KEEP
+    assert spans[0].context.upstream_services == "dW5uYW1lZC1weXRob24tc2VydmljZQ|2|4|"
```

Comment:
I couldn't tell if this is the right place for this test/assertion, or if I should add specific manual keep/drop test cases to `test_upstream_services.py`. Open to opinions here.